File: src/Client/mocks/company.ts
import { faker } from '@faker-js/faker'
import type { Company } from '~/routes/__layout/companies/index'

export function buildCompany(): Company {
  return {
    address: faker.address.streetAddress(),
    ceo: faker.name.fullName(),
    contacts: faker.internet.email(),
    email: faker.internet.email(),
    id: faker.datatype.number(),
    inn: faker.random.numeric(10),
    name: faker.company.name(),
    phone: faker.phone.number(),
    type: faker.helpers.arrayElement(["ООО", "АО", "ПАО", "ИП"])
  }
}

File: src/Domain/Entities/Company.cs
﻿namespace CRM.Domain.Entities;

public class Company : BaseAuditableEntity
{
    public int? TypeId { get; set; }
    public virtual CompanyType? Type { get; set; }
    public string Name { get; set; } = default!;
    public string? Inn { get; set; }
    public string? Address { get; set; }
    public string? Ceo { get; set; }
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string? Contacts { get; set; }
    public string? ManagerId { get; set; }
    public virtual ApplicationUser? Manager { get; set; }
}
